lib/src/main/java/io/ably/lib/http/HttpCore.java

```diff
@@ -77,21 +114,21 @@ public HttpCore(ClientOptions options, Auth auth, PlatformAgentProvider platform
      */
     public <T> T httpExecuteWithRetry(URL url, String method, Param[] headers, RequestBody requestBody, ResponseHandler<T> responseHandler, boolean requireAblyAuth) throws AblyException {
         boolean renewPending = true, proxyAuthPending = true;
-        if(requireAblyAuth) {
+        if (requireAblyAuth) {
             authorize(false);
         }
-        while(true) {
+        while (true) {
             try {
                 return httpExecute(url, getProxy(url), method, headers, requestBody, true, responseHandler);
-            } catch(AuthRequiredException are) {
-                if(are.authChallenge != null && requireAblyAuth) {
-                    if(are.expired && renewPending) {
+            } catch (AuthRequiredException are) {
```

Comment:
Exception message should be logged. This is true for all other exceptions in other places. Helpful for debugging